importers/wecdis_importer.py

```diff
@@ -0,0 +1,493 @@
+
+        contact_a = self.generate_ttm_contact(
+            sensor_name + "_a",
+            data_store,
+            datafile,
+            line_number,
+            detecting_platform,
+            range_a_token,
+            bearing_a_token,
+            change_id,
+        )
+        contact_b = self.generate_ttm_contact(
+            sensor_name + "_b",
+            data_store,
+            datafile,
+            line_number,
+            detecting_platform,
+            range_b_token,
+            bearing_b_token,
+            change_id,
+        )
+
+        location = self._parse_lat_lon_tokens(lat_token, lat_hem_token, lon_token, lon_hem_token)
+        if location:
+            contact_a.location = location
+            contact_b.location = location
+
+        combine_tokens(lat_token, lon_token).record(self.name, "location", location, "DMS")
+
+        contact_a.track_number = f"TTM_{sensor_name}_a"
+        contact_b.track_number = f"TTM_{sensor_name}_b"
+
+    def generate_ttm_contact(
+        self,
+        sensor_name,
+        data_store,
+        datafile,
+        line_number,
+        detecting_platform,
+        range_token,
+        bearing_token,
+        change_id,
+    ):
+        """Generates a contact from a TTM sensor range/bearing
+        TTM has two ranges and bearings - appearing to come from two linked sources
+        :param sensor_name: The name of the sensor that we're generating this contact from
+        :param data_store: The data store that this is importing into
+        :param datafile: The datafile being imported
+        :param line_number: The number of the line currently being processed
+        :param detecting_platform: The platform that's detecting the contact
+        :param range_token: The token giving the range of the contact
+        :param bearing_token: The token giving the bearing of the contact
+        :param change_id: The ID representing this import as a change
+        """
+        detecting_sensor = self.get_cached_sensor(
+            data_store,
+            sensor_name=f"TTM-{sensor_name}",
+            sensor_type=f"{sensor_name}",
+            platform_id=detecting_platform.platform_id,
+            change_id=change_id,
+        )
+        contact = datafile.create_contact(
+            data_store=data_store,
+            platform=detecting_platform,
+            sensor=detecting_sensor,
+            timestamp=self.timestamp,  # This contact type doesn't have its own timestamp
+            parser_name=self.short_name,
+        )
+
+        bearing_valid, bearing = convert_absolute_angle(
+            bearing_token.text, line_number, self.errors, self.error_type
+        )
+        if bearing_valid:
+            contact.bearing = bearing
+            bearing_token.record(self.name, "bearing", bearing)
+
+        range_valid, contact_range = convert_distance(
+            range_token.text,
+            unit_registry.kilometers,  # TODO - confirm whether units are km, kyds or NM
```

Comment:
this is certainly quite a biggie

Reply:
No units are mentioned in MWC's parser, how does Debrief handle the REP data that's generated? Can the analysts update the units in Debrief after the REP file is loaded?